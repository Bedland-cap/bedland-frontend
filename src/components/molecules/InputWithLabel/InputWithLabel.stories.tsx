import type { Meta, StoryObj } from '@storybook/react';
import { useForm, FieldValues, UseFormRegister } from 'react-hook-form';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import loginRegexPattern from 'components/organisms/LoginForm/loginRegexPatterns';
import InputWithLabel from './InputWithLabel';

const meta = {
  title: 'Basic Components/Molecules/InputWithLabel',
  component: InputWithLabel,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof InputWithLabel>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: (args) => {
    const { register } = useForm();
    return (
      <form>
        <InputWithLabel {...args} register={register} />
      </form>
    );
  },
  args: {
    input: 'login',
    type: 'text',
    placeholder: 'Your login',
    label: 'Login',
    register: {} as UseFormRegister<FieldValues>,
    regexPattern: loginRegexPattern.login,
  },
};
