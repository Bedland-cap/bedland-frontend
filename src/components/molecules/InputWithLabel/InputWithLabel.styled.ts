import styled from 'styled-components';

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
`;

export default InputContainer;
