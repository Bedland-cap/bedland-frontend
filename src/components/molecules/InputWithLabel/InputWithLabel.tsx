import { FieldValues, UseFormRegister } from 'react-hook-form';
import Input from 'components/atoms/Input/Input';
import Label from 'components/atoms/Label/Label';
import InputContainer from './InputWithLabel.styled';

export interface IInputWithLabel {
  label: string;
  input: string;
  type: string;
  placeholder: string;
  register: UseFormRegister<FieldValues>;
}

const InputWithLabel = ({
  label,
  input,
  type = 'text',
  placeholder,
  register,
}: IInputWithLabel) => (
  <InputContainer>
    <Label htmlFor={input} label={label} textColor='white' size='medium' />
    <Input
      input={input}
      placeholder={placeholder}
      register={register}
      type={type}
    />
  </InputContainer>
);

export default InputWithLabel;
