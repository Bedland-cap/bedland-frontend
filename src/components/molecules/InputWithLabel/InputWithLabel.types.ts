import { IInput } from 'components/atoms/Input/Input.types';

export interface IInputWithLabel extends IInput {
  label: string;
}
