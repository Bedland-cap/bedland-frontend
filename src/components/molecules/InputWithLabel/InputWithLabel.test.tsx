import renderer from 'react-test-renderer';
import { useForm } from 'react-hook-form';
import ThemeProvider from 'theme/ThemeContext';
import InputWithLabel from './InputWithLabel';

const SnapshotInput = () => {
  const { register } = useForm();
  return (
    <ThemeProvider>
      <form action=''>
        <InputWithLabel
          label='login'
          input='login'
          type='text'
          placeholder='login'
          register={register}
        />
      </form>
    </ThemeProvider>
  );
};

describe('Input with Label', () => {
  it('renders correctly', () => {
    const snapshot = renderer.create(<SnapshotInput />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
