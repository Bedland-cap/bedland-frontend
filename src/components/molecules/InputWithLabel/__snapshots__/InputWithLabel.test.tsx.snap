// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input with Label renders correctly 1`] = `
.c2 {
  border: 1px solid #909AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c2:disabled,
.c2:hover,
.c2:active,
.c2:focus {
  outline: none;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
}

<form
  action=""
>
  <div
    className="c0"
  >
    <label
      className="c1"
      htmlFor="login"
      size="medium"
    >
      login
    </label>
    <input
      className="c2"
      id="login"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login"
      type="text"
    />
  </div>
</form>
`;
